specs/StatusBarMessage.md

```diff
@@ -0,0 +1,106 @@
+<!-- USAGE
+  * Fill in each of the sections (like Background) below
+  * Wrap code with `single line of code` or ```code block```
+  * Before submitting, delete all <!-- TEMPLATE marked comments in this file,
+    and the following quote banner:
+-->
+# Background
+The browser has a Status bar that displays text when hovering over a link, or performing some activity. Currently, 
+developers are able to opt in to disable showing the status bar through the browser 
+settings.
+
+Developers would also like to be able to opt in to intercept the messages which would 
+normally be displayed by the Status bar, and show it using thier own custom UI. 
+# Description
+We propose a new event for WebView2 that would allow developers to 
+listen for Status bar updates which are triggered by activity on the WebView, and then handle those updates however they want in their applications.
+
+Developers will be able to register an event handler for changes to the status bar message.
+# Examples
+## Win32 C++ Registering a listener for status bar message changes
+```
+CHECK_FAILURE(m_webView->add_StatusBarMessageChanged(
+    Microsoft::WRL::Callback<ICoreWebView2StatusBarMessageChangedEventHandler>(
+    [this](ICoreWebView2* sender, IUnknown* args) -> HRESULT
+    {
+        
+        LPWSTR value;
+        CHECK_FAILURE(sender->get_StatusBarMessage(&value));
```

Comment:
Does returning an HR actually do anything in the sender? If not, don't want to imply it in the sample code.

Reply:
This should be our sample app's macro for checking the HRESULT, showing a message box, and terminating the process. Should be OK to be consistent with that.

Reply:
Ah, I didn't realize it was terminating. Thanks.